Allow middle click on notification to invoke action

miseran opened the pull request:
A middle click on a notification with a single or default action will
invoke it. If there are multiple actions and no default, the context
menu is opened. If there are no actions, proceed similarly with URLs.

As discussed in #234.

I decided to always prioritize actions over URLs, since that seems more likely to be what the user wants.

tsipinakis commented on src/notification.c:
```diff
@@ -588,4 +588,32 @@ int notification_get_ttl(notification *n) {
+
+        } else if (n->urls) {
+                if (strstr(n->urls, "\n") == NULL)
+                        open_browser(n->urls);
```
Apparently, urls that were parsed from markup tags (e.g. `<a href="http://www.google.com">Link</a>`), they are in the form of `[#n] url`.

tsipinakis commented:
And as a (optional) sidenote: Unless there's a reason for it, it's usually considered good etiquette to have your name and/or a valid email address in your commit. See the [git documentation](https://git-scm.com/book/tr/v2/Customizing-Git-Git-Configuration)

miseran commented:
- The button release passing to the window below also happened with left and right clicks, as I found out with xev. Hence, I changed all mouse clicks to be on release.
- I don't have a solution to the problem with multiple middle clicks either. It doesn't happen with rofi, which closes if you click anywhere else.
- I moved the code that removes the `[#1]` prefix to the open_browser function, so it applies to middle clicks as well. In fact, it appears `<a href>` links were actually broken and would crash dunst, which seems to be fixed with my commit. Apparently it was a bad free, but I don't understand this very well.

I should say that I'm not very experienced with C, but I hope my code is acceptable anyway.

tsipinakis commented:
> In fact, it appears `<a href>` links were actually broken and would crash dunst, which seems to be fixed with my commit. Apparently it was a bad free, but I don't understand this very well.

I had this issue in mind but didn't get around to fixing it yet, guess this resolves that too.

Looking at how the code was previously, looks like it's because the `in` pointer is re-assigned to the result of the `strchr` call and it crashes since it's an error to call `free` with a pointer that doesn't point to the beginning of an allocated block.

> I moved the code that removes the `[#1]` prefix to the open_browser function, so it applies to middle clicks as well.
> I should say that I'm not very experienced with C, but I hope my code is acceptable anyway.

I *really* don't like this solution but it's not a problem with your code, it's a problem with the implementation of urls and actions in dunst generally, and looking at it, this is probably the best that can be done right now. I'm slowly working towards refactoring it into a simpler and hopefully more expandable system but it takes time as usual.

I'm merging this as-is then. Thank you for your work.
